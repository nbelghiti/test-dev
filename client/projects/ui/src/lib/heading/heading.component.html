<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
