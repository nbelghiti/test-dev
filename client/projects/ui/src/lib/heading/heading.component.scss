// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
