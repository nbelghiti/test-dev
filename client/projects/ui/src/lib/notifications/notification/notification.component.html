<div
	class="c-notification"
	[ngClass]="{
		'c-notification--success': notificationType.success === type,
		'c-notification--error': notificationType.error === type
	}"
>
	<ui-row contentAfter="true">
		<div class="c-notification__title">
			<div class="c-notification__icon">
				<ui-icon
					[name]="icon"
					size="tiny"
				>
				</ui-icon>
			</div>

			<ui-heading
				[level]="3"
				[title]="title | translate"
			>
			</ui-heading>
		</div>

		<span
			contentAfter
			class="c-notification__time"
			*ngIf="parsedTime"
		>
			{{ parsedTime }}
		</span>
	</ui-row>

	<div class="c-notification__content">
		<ng-content></ng-content>
	</div>

	<button
		*ngIf="!isRead"
		class="c-notification__mark"
		(click)="markRead()"
	>
		Mark as read
	</button>
</div>
