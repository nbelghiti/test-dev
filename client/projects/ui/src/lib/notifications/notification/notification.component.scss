// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$icon-size: 2.4rem;

// =============================================================================
// :: Notification
// =============================================================================
.c-notification {
	@include extend-list-clear();

	padding: 2.5rem 3rem 2rem;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-notification__title {
	display: flex;
	align-items: center;
}

.c-notification__icon {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $icon-size;
	height: $icon-size;
	margin-right: 0.8rem;

	line-height: $icon-size;

	border-radius: 50%;
	background-color: $background-color-grey-96;
}

.c-notification__time {
	font-size: $font-size-1x-small;

	color: $text-color-grey-56;
}

.c-notification__content {
	margin-top: 1rem;
}

.c-notification__mark {
	width: 100%;

	margin-top: 1rem;

	text-align: right;

	color: $primary-color;
}

// =============================================================================
// :: States
// =============================================================================
.c-notification--success {

	.c-notification__icon {
		color: $ui-color-success-dark;
		background-color: $ui-color-success-light;
	}
}

.c-notification--error {

	.c-notification__icon {
		color: $ui-color-error-dark;
		background-color: $ui-color-error-light;
	}
}
