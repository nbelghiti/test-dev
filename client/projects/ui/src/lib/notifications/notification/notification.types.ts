export interface INotificationReadStatus {
	notificationId: string;
	isRead: boolean;
}
