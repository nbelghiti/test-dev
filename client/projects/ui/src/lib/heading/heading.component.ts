import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
