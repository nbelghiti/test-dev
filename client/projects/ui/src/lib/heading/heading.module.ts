import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
