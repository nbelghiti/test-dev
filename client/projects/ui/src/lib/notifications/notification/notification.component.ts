import { AfterContentInit, Component, EventEmitter, Input, Output } from '@angular/core';

import { IconName } from '../../icon/icon.types';
import { NotificationType } from '../notifications.types';

import { INotificationReadStatus } from './notification.types';

@Component({
	selector: 'ui-notification',
	templateUrl: './notification.component.html',
	styleUrls: ['./notification.component.scss']
})

export class NotificationComponent implements AfterContentInit {
	@Input() public notificationId: string;
	@Input() public isRead: boolean;
	@Input() public title: string;
	@Input() public time: string;
	@Input() public type: NotificationType = NotificationType.success;
	@Input() public icon: IconName = IconName.exclamationCircle;

	@Output() public notificationReadStatus: EventEmitter<INotificationReadStatus> = new EventEmitter<INotificationReadStatus>();

	public notificationType = NotificationType;
	public parsedTime: string;

	public ngAfterContentInit(): void {
		const locale = 'nl-BE';

		if (!this.time) {
			return;
		}

		const timeStamp = new Date(this.time);
		const time = timeStamp.toLocaleTimeString(locale, {
			hour: 'numeric',
			minute: '2-digit'
		});
		const date = timeStamp.toLocaleDateString(locale);

		this.parsedTime = `${this.isToday(timeStamp) ? 'Today' : date.replace(/\//g, '-')} - ${time}`;
	}

	public isToday = (someDate: Date): boolean => {
		const today = new Date();
		return someDate.getDate() === today.getDate() &&
			someDate.getMonth() === today.getMonth() &&
			someDate.getFullYear() === today.getFullYear();
	}

	public markRead(): void {
		this.notificationReadStatus.emit({
			notificationId: this.notificationId,
			isRead: true,
		});
	}
}
