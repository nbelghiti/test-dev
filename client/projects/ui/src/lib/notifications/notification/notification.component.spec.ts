import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';

import { HeadingComponent } from '../../heading/heading.component';
import { IconComponent } from '../../icon/icon.component';
import { RowComponent } from '../../row/row.component';

import { NotificationComponent } from './notification.component';

describe('NotificationComponent', () => {
	let component: NotificationComponent;
	let fixture: ComponentFixture<NotificationComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				TranslateModule.forRoot(),
			],
			declarations: [
				HeadingComponent,
				IconComponent,
				NotificationComponent,
				RowComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(NotificationComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
