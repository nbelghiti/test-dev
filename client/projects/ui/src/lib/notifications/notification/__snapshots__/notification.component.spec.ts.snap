// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent Should render correctly 1`] = `
<ui-notification
  icon={[Function String]}
  isToday={[Function Function]}
  notificationReadStatus={[Function EventEmitter]}
  notificationType={[Function Object]}
  type={[Function String]}
>
  <div
    class="c-notification c-notification--success"
    ng-reflect-klass="c-notification"
    ng-reflect-ng-class="[object Object]"
  >
    <ui-row
      contentafter="true"
      ng-reflect-content-after="true"
    >
      <div
        class="c-row"
        ng-reflect-klass="c-row"
        ng-reflect-ng-class="[object Object]"
      >
        
        <div
          class="c-row__content"
        >
          <div
            class="c-notification__title"
          >
            <div
              class="c-notification__icon"
            >
              <ui-icon
                class="c-icon"
                ng-reflect-name="exclamation-circle"
                ng-reflect-size="tiny"
                size="tiny"
              >
                <i
                  class="c-icon__glyph c-icon__glyph--tiny  fas fa-exclamation-circle"
                />
              </ui-icon>
            </div>
            <ui-heading
              ng-reflect-level="3"
            >
              <div
                class="c-heading"
                ng-reflect-klass="c-heading "
                ng-reflect-ng-class="[object Object]"
              >
                
                
                
                
                <h3>
                  
                </h3>
                <div
                  class="c-heading__meta"
                >
                  
                  
                  
                </div>
              </div>
            </ui-heading>
          </div>
        </div>
        
        <div
          class="c-row__after"
        >
          
        </div>
      </div>
    </ui-row>
    <div
      class="c-notification__content"
    />
    
    <button
      class="c-notification__mark"
    >
       Mark as read 
    </button>
  </div>
</ui-notification>
`;
